generals/core/observation.py

```diff
@@ -44,43 +48,48 @@ def values(self):
     def items(self):
         return dataclasses.asdict(self).items()
 
-    def as_tensor(self, pad_to: int | None = None) -> np.ndarray:
+    def to_tensor(self, pad_to: int = None) -> np.ndarray:
         """
-        Returns a 3D tensor of shape (15, rows, cols). Suitable for neural nets.
+        Returns a uniformly sized tensor with shape: (15, height, width).
+        Scalar fields such as timestep or opponent_land_count are
+        upsampled to the size of the grid.
+
+        Optionally increases the size of channels to pad_to, i.e.
+        the returned shape becomes: (15, pad_to).
         """
-        shape = self.armies.shape
+
+        grid_dims = self.armies.shape
+
         if pad_to is not None:
-            shape = (pad_to, pad_to)
-            assert pad_to >= max(self.armies.shape), "Can't pad to a smaller size than the original observation."
-            # pad every channel with zeros, except for mountains, those are padded with ones
-            h_pad = (0, pad_to - self.armies.shape[0])
-            w_pad = (0, pad_to - self.armies.shape[1])
-            self.armies = np.pad(self.armies, (h_pad, w_pad), "constant")
-            self.generals = np.pad(self.generals, (h_pad, w_pad), "constant")
-            self.cities = np.pad(self.cities, (h_pad, w_pad), "constant")
-            self.mountains = np.pad(self.mountains, (h_pad, w_pad), "constant", constant_values=1)
-            self.neutral_cells = np.pad(self.neutral_cells, (h_pad, w_pad), "constant")
-            self.owned_cells = np.pad(self.owned_cells, (h_pad, w_pad), "constant")
-            self.opponent_cells = np.pad(self.opponent_cells, (h_pad, w_pad), "constant")
-            self.fog_cells = np.pad(self.fog_cells, (h_pad, w_pad), "constant")
-            self.structures_in_fog = np.pad(self.structures_in_fog, (h_pad, w_pad), "constant")
```

Comment:
Just a heads up, `np.pad`'s API is kinda confusing/odd! It adds the specified padding to each end of the dimension. See example:
```
>>> import numpy as np
>>> arr = np.ones((5,5))
>>> arr.shape
(5, 5)
>>> padded_arr = np.pad(arr, (2,2))
>>> padded_arr.shape
(9, 9)
>>>
```